.github/actions/setup-python/action.yml

```diff
@@ -0,0 +1,65 @@
+name: Setup Python and install requirements
+inputs:
+  python-version:
+    required: true
+  architecture:
+    default: x64
+runs:
+  using: composite
+  steps:
+    # TODO: GH#44980 https://github.com/pypa/setuptools/issues/2941
+    - name: Create temporary requirements.txt
+      run: |
+        # Drop cache at least once per month
+        month_today="$(date '+%Y-%m')"
+        cat > requirements.txt <<EOF
+          # $month_today
+
+          # Python deps
+          pip
+          setuptools<60.0.0
```

Comment:
Hmm don't love that these deps (pandas + test) are also specified in yaml files as well (aware over multiple yaml files as well). Any idea on how to combine them?

Reply:
Before the refactoring they were part of an inline `pip install ...`.

We might be able to auto-generate them from a minimal `environment.yml` file but not sure if that's better than having a little duplication.

Reply:
Gotcha. Could this be maintained in a separate file and modified here with the date to invalidate the cache?

Reply:
Happy to do it, but we'd still need to add the current timestamp (or something derived from it, like current month) to the file because otherwise the cache will only be invalidated when we change the file, which we don't do on a regular basis.

Reply:
NB: For `setup-python` a stale cache just means slower installs, not older versions.

Reply:
Yeah makes sense; this block can create a copy of the environment file and add a date. I think it's just easier to manage/know that our dependencies are in files and not as ad hoc like this.